packages/forms/src/FieldArray.tsx

```diff
@@ -1,42 +1,31 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import * as R from 'ramda';
 import { FieldArray as FinalFieldArray } from 'react-final-form-arrays';
-import { compose, setDisplayName } from 'recompose';
 import { FieldArrayProps } from 'react-final-form-arrays';
+import { tableSelectors } from '@8base/utils';
 
-import { withFieldSchema, WithFieldSchemaProps } from './utils';
-
-const enhacner: any = compose(
-  withFieldSchema,
-  setDisplayName('FieldArray'),
-);
+import { FormContext } from './FormContext';
+import { getFieldSchemaName } from './utils';
 
 /**
  * `FieldArray` wrapper based on `FieldArray` from the [`react-final-form-arrays`](https://github.com/final-form/react-final-form-arrays). It accepts [`FieldArrayProps`](https://github.com/final-form/react-final-form-arrays#fieldarrayprops) props.
  */
-const FieldArray: React.ComponentType<FieldArrayProps<any, any>> = enhacner(
-  class FieldArray extends React.Component<FieldArrayProps<any, any> & WithFieldSchemaProps> {
-    public collectProps = () => {
-      const { fieldSchema, name, isEqual, ...restProps } = this.props;
-
-      if (fieldSchema) {
-        return {
-          fieldSchema,
-          isEqual: isEqual || R.equals,
-          name: name || fieldSchema.name,
-          ...restProps,
-        };
-      }
-
-      return this.props;
-    };
-
-    public render() {
-      const collectedProps = this.collectProps();
-
-      return <FinalFieldArray {...collectedProps} />;
+const FieldArray = (props: FieldArrayProps<any, any>) => {
+  const { tableSchema } = useContext(FormContext);
```

Comment:
@zouxuoz Why do you use here `FomrContext` instead of `TableSchemaContext`?

Reply:
@vorobeez because we pass `tableSchemaName` only to the `Form`, if we want to use only `TableSchemaContext` we should pass `tableSchemaName` to the each `Field`